### PR TITLE
Small corrections and additions to migration guide 2.5

### PR DESCRIPTION
**Pull request by marcospereira**
1. Replace `Optional.ifPresent` suggestion with `Optional.orElse`
2. Add links to javadocs of Optional methods
3. Add some missing punctuation

### CODE REVIEW

File: documentation/manual/releases/Migration25.md [hunk 1]
```diff
@@ -53,39 +53,39 @@ If you are using the java8 lambda syntax and the lambda's body does not throw a
 
 Last but not least, if you are using `F.Callback3<A,B,C>` in your application, since there is no Java8 replacement for it, you may want to use `akka.japi.function.Function3`.
 
-## Replaced `F.Option` with Java8 `Optional`
+## Replaced `F.Option` with Java8 `java.util.Optional`
 
-`F.Option` has been discontinued and we embraced the Java8 `Optional` type. The two types are similar, but their API is different, so you will need to update your code. The main difference between the two types is that while `F.Option` inherited from `java.util.Collection`, `Optional` doesn't.
+`F.Option` has been discontinued and we embraced the Java8 `Optional` type. The two types are similar, but their API is different, so you will need to update your code. The main difference between the two types is that while `F.Option` inherited from `java.util.Collection`, `java.util.Optional` doesn't.
```
Comment (gmethvin) [inline documentation/manual/releases/Migration25.md]: Cane you make the first reference to `F.Option` and `Optional` links to the API docs here? I know you have a link below but I think linking the first reference makes it more convenient if you're skimming the docs.

File: documentation/manual/releases/Migration25.md [hunk 2]
```diff
@@ -53,39 +53,39 @@ If you are using the java8 lambda syntax and the lambda's body does not throw a
...
+If you are using `controllers.ExternalAssets` in your routes file you must either set `routesGenerator := InjectedRoutesGenerator` in your `build.sbt` or you must use the `@` symbol in front of the route like `GET /some/path @controllers.ExternalAssets.at`.
 
 ## Removed Plugins API
 
-Play's Plugin API was deprecated in 2.4, and has been removed.  
+Play's Plugin API was deprecated in 2.4, and has been removed.
 
-To create reusable components that are dependency injection independent, please use Play's module system [`play.api.inject.Module`](api/scala/play/api/inject/Module.html).  Otherwise, a singleton bound to a dependency injected module (either [[Scala|ScalaDependencyInjection]] or [[Java|JavaDependencyInjection]]) is usually enough to replace the Plugin API completely.
+To create reusable components that are Dependency Injection independent, please use Play's module system [`play.api.inject.Module`](api/scala/play/api/inject/Module.html).  Otherwise, a singleton bound to a Dependency Injected module (either [[Scala|ScalaDependencyInjection]] or [[Java|JavaDependencyInjection]]) is usually enough to replace the Plugin API completely.
```
Comment (gmethvin) [inline documentation/manual/releases/Migration25.md]: What's the significance of capitalizing Dependency Injection here? 

This should probably say something like:
> You should use dependency injection to bind Play components. `onStart` should be replaced by initialization in your component's constructor and `onStop` by calling the `addStopHook` in [`ApplicationLifecycle`](api/scala/play/api/inject/ApplicationLifecycle.html).  See the dependency injection documentation for [[Scala|ScalaDependencyInjection]] or [[Java|JavaDependencyInjection]] for more details.

File: documentation/manual/releases/Migration25.md (hunk 1 above)
Comment (marcospereira) [inline documentation/manual/releases/Migration25.md]: Done.

File: documentation/manual/releases/Migration25.md (hunk 2 above)
Comment (marcospereira) [inline documentation/manual/releases/Migration25.md]: Removed the capitalization.

**Comment by cchantep** [comment]
Lgtm

**Comment by cchantep** [comment]
Need a rebase I guess

**Comment by marcospereira** [comment]
This PR looks irrelevant now, considering the changes made at #5554.

I'm closing it.

**Pull request closed without merging**